vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_authorization.cc

```diff
@@ -229,12 +215,95 @@ void UpholdAuthorization::OnCardCreate(
             address.substr(0, 5));
   }
 
-  base::flat_map<std::string, std::string> args;
   if (wallet_ptr->status != type::WalletStatus::VERIFIED) {
+    base::flat_map<std::string, std::string> args{};
     args["redirect_url"] = GetSecondStepVerify();
+    callback(type::Result::LEDGER_OK, args);
   }
 
-  callback(type::Result::LEDGER_OK, args);
+  GetAnonFunds(
+      std::bind(&UpholdAuthorization::OnGetAnonFunds, this, _1, _2, callback));
+}
+
+void UpholdAuthorization::GetAnonFunds(
+    endpoint::promotion::GetWalletBalanceCallback callback) const {
+  // if we don't have user funds in anon card anymore
+  // we can skip balance server ping
+  if (!ledger_->state()->GetFetchOldBalanceEnabled()) {
+    callback(type::Result::LEDGER_OK, type::Balance::New());
+    return;
+  }
+
+  const auto wallet = ledger_->wallet()->GetWallet();
+  if (!wallet) {
+    BLOG(1, "Wallet is not created!");
+    ledger_->state()->SetFetchOldBalanceEnabled(false);
+    callback(type::Result::LEDGER_OK, type::Balance::New());
+    return;
+  }
+
+  if (wallet->payment_id.empty()) {
+    BLOG(0, "Payment ID is empty!");
```

Comment:
Definitely not something we should address with this PR, but I'm curious (@zenparsing, @emerick) if it would make sense to switch to using constants for the info levels?  
https://github.com/brave/brave-core/blob/5a4a600a21450c7764751a7b3cc9ff56eafe654d/vendor/bat-native-ledger/src/bat/ledger/internal/logging/logging.h#L45-L53

We could address by making an enum
```cpp
enum LedgerLogLevel {
    Error,
    Info,
    UrlRequest = 5,
    UrlResponse,
    UrlResponseWithBody,
    DbQueries
};
// ...
if (wallet->payment_id.empty()) {
    BLOG(LedgerLogLevel::Error, "Payment ID is empty!");
    // ...
```

Reply:
This would be a lot more informative.

Reply:
I definitely like that idea.